cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,595 @@
+     * 6. Overrides DurabaleDataLog Metadata with the highest epoch seen.
+     * @param containerId Id of the container to be created.
+     * @param allFiles all chunk files to be parsed.
+     * @param context Context holding config.
+     * @param dataLogFactory Factory of DurabeleDataLog to be created.
+     * @throws Exception
+     */
+    private void recoverFromStorage(int containerId, File[] allFiles, Context context, DurableDataLogFactory dataLogFactory ) throws Exception {
+        File[] potentialFiles = getContainerMetadataChunkFiles(allFiles, containerId);
+        deleteJournalsOf(containerId);
+        //create debug segment container
+        DebugStreamSegmentContainer debugStreamSegmentContainer = createDebugSegmentContainer(context, containerId, dataLogFactory);
+        output("-----------------DebugSegment container %d initialized--------------", containerId);
+        // segregate the metadata and storage metadata chunks
+        Map<String, List<File>> metadataSegments = segregateMetadataSegments(potentialFiles);
+        Preconditions.checkState(metadataSegments.size() == 2, "Only MetadataSegment and Storage MetadataSegment chunks should be present");
+        List<File> storageChunks = null;
+        boolean firstRun = true;
+        int attempts = 0;
+
+        ChunkValidator chunkValidator = new ChunkValidator(debugStreamSegmentContainer);
+        Map<Integer, Long> containersToEpoch = new HashMap<>();
+        while (firstRun || (!chunkValidator.validate() /*&& attempts < RETRY_ATTEMPT*/)) {
+            if (firstRun) {
+                firstRun = !firstRun;
+            }
+            for (Map.Entry<String, List<File>> segmentEntries : metadataSegments.entrySet()) {
+                List<File> chunks = segmentEntries.getValue();
+                chunks.sort(new FileComparator());
+                setEpochforContainer(chunks.get(chunks.size() - 1), containerId, containersToEpoch);
+                if (chunks.get(0).getName().contains("storage")) {
+                    storageChunks = chunks;
+                    continue;
+                }
+                List<TableSegmentUtils.TableSegmentOperation> tableSegmentOperations = TableSegmentUtils.getOperationsFromChunks(chunks);
+                writeEntriesToContainerMetadata(debugStreamSegmentContainer, NameUtils.getMetadataSegmentName(containerId), tableSegmentOperations);
+            }
+            //write storage entries in the end after container entries (to avoid length mismatch).
+            if (storageChunks != null) {
+                writeEntriesToStorageMetadata(debugStreamSegmentContainer, NameUtils.getStorageMetadataSegmentName(containerId), TableSegmentUtils.getOperationsFromChunks(storageChunks));
+            }
+            //flush to storage
+            output("Flushing to storage");
+            flushToStorage(debugStreamSegmentContainer);
+            attempts++;
```

Comment:
looks like attempts is not used

Reply:
part of the RETRY_ATTEMPT explained above.

Reply:
Uncommented it to introduce retries now with the latest set of changes to avoid confusion.